be/src/storage/lake/tablet.cpp

```diff
@@ -58,8 +58,22 @@ StatusOr<std::unique_ptr<TabletWriter>> Tablet::new_writer() {
 }
 
 StatusOr<std::shared_ptr<const TabletSchema>> Tablet::get_schema() {
-    ASSIGN_OR_RETURN(auto metadata, get_metadata(1));
-    return GlobalTabletSchemaMap::Instance()->emplace(metadata->schema()).first;
+    auto tablet_schema_key = fmt::format("{}", _id);
+    auto ptr = _mgr->lookup_tablet_schema(tablet_schema_key);
+    RETURN_IF(ptr != nullptr, ptr);
+
+    ASSIGN_OR_RETURN(TabletMetadataIter metadata_iter, _mgr->list_tablet_metadata(_group, _id));
+    if (!metadata_iter.has_next()) {
+        return Status::NotFound(fmt::format("tablet {} metadata not found", _id));
+    }
+    ASSIGN_OR_RETURN(auto metadata, metadata_iter.next());
+    ptr = GlobalTabletSchemaMap::Instance()->emplace(metadata->schema()).first;
+    if (ptr == nullptr) {
+        return Status::InternalError(fmt::format("tablet schema {} failed to emplace in TabletSchemaMap", _id));
+    }
+    auto value_ptr = new std::shared_ptr<const TabletSchema>(ptr);
+    (void)_mgr->fill_metacache(tablet_schema_key, static_cast<void*>(value_ptr), metadata->schema().SpaceUsedLong());
```

Comment:
1. should check if the returned TabletSchema is a shared object and if so the memory cost should be zero.
2. call `TabletSchema::mem_usage()` instead of `TabletSchemaPB::SpaceUsedLong()`